package/kedro_viz/launchers/jupyter.py

```diff
@@ -89,15 +91,53 @@ def _allocate_port(start_at: int, end_at: int = 65535) -> int:
     socket.setdefaulttimeout(2.0)  # seconds
     for port in acceptable_ports:  # iterate through all acceptable ports
         with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as sock:
-            if sock.connect_ex(("127.0.0.1", port)) != 0:  # port is available
+            if sock.connect_ex((host, port)) != 0:  # port is available
                 return port
 
     raise ValueError(
         "Cannot allocate an open TCP port for Kedro-Viz in a range "
-        "from {} to {}".format(start_at, end_at)
+        f"from {start_at} to {end_at}"
     )
 
 
+def _is_databricks() -> bool:
+    return "DATABRICKS_RUNTIME_VERSION" in os.environ
+
+
+def _get_databricks_object(name: str):
+    """Gets object called `name` from the user namespace."""
+    return IPython.get_ipython().user_ns.get(name)
+
+
+def _make_databricks_url(port: int) -> str:
+    """Generates the URL to the Kedro-Viz instance."""
+    dbutils = _get_databricks_object("dbutils")
+
+    if dbutils is None:
+        raise EnvironmentError("Unable to find dbutils.")
+
+    def dbutils_get(attr):
+        return getattr(
+            dbutils.notebook.entry_point.getDbutils().notebook().getContext(), attr
+        )().get()
+
+    browser_host_name = dbutils_get("browserHostName")
+    workspace_id = dbutils_get("workspaceId")
+    cluster_id = dbutils_get("clusterId")
+    path_name = f"/driver-proxy/o/{workspace_id}/{cluster_id}/{port}/"
+
+    return f"https://{browser_host_name}{path_name}"
+
+
+def _display_databricks_html(port: int):
+    url = _make_databricks_url(port)
+    displayHTML = _get_databricks_object("displayHTML")  # pylint: disable=invalid-name
+    if displayHTML is not None:
+        displayHTML(f"""<a href="{url}">Open Kedro-Viz</a>""")
+    else:
+        print(f"Kedro-Viz is available at {url}")
+
+
 # pylint: disable=unused-argument,missing-type-doc
 def run_viz(port: int = None, line=None, local_ns=None) -> None:
```

Comment:
The `local_ns` is not used anywhere.

Reply:
Are `line` and `local_ns` required in the function signature so that this works as a line magic, though they are unused?

Reply:
Yeah, I think what @jmholzer says here is right. I tried removing them before and IPython raised some sort of "invalid line magic" exception.

Reply:
This isn't documented anywhere, but I think this is the bit why `local_ns` is needed, I am not sure what it does. This is also an unofficial way to extend `kedro`'s line magic by adding things into `kedro.line_magic` in `setup.py`

```
    for line_magic in load_entry_points("line_magic"):
        register_line_magic(needs_local_scope(line_magic))
        logger.info("Registered line magic '%s'", line_magic.__name__)  # type: ignore
```

Reply:
Ah yes, you're right. The `local_ns` is required due to the `needs_local_scope` on the kedro ipython extension side. This means _any_ line magics registered with the `kedro.line_magic` entrypoint would need to have this argument.

Previously this was actually used to get the project path as in `local_ns["project_path"]`. In fact, I'd hazard a guess that was why `needs_local_scope` was added in the kedro ipython extension in the first place. Definitely doesn't feel right to me that we're forcing all line magics registered with `kedro.line_magic` to require that argument. And, as you say, this is completely undocumented (as is the existence of that entrypoint to begin with...)

To be honest, now I think about it again I'm not sure why we even expose this entrypoint rather than just letting extensions (like kedro-mlflow, which is probably the only thing besides kedro-viz that has ever used this entrypoint) register their own line magics as in `%load_ext kedro_mlflow`. I guess it's slightly more convenient since then you don't need to do multiple `%load_ext`, but it feels a bit weird to me... Let me make a separate issue for it on the kedro repo anyway.

Reply:
@noklam I've removed `line` as an argument since it wasn't doing anything. In also looked like you could do `%run_viz 5000` before to set a custom port, but actually that wouldn't have worked because `port` would be the string `"5000"`.

We've never advertised `%run_viz` as taking arguments (and until now this wouldn't have worked). I'm not going to document or advertise it because I don't want to have to maintain it as a feature, since ultimately we probably want `%run_viz` to properly handle many arguments in the same sort of way that `%reload_kedro` can now. It might be handy to have this available e.g. if databricks start locking down ports of something. So let's bear in mind in case anyone wants to do it on databricks, for now at least you can do `%run_viz 5000`... but this is not officially condoned since it's kind of an experimental feature anyway 😀

Reply:
FYI @jmholzer ☝️

Reply:
Also FYI for @noklam: https://github.com/kedro-org/kedro/issues/878.